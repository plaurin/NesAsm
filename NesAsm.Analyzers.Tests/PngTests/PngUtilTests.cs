using ApprovalTests;
using ApprovalTests.Reporters;
using BigGustave;
using System.Text;

namespace NesAsm.Analyzers.Tests.PngTests;

[UseReporter(typeof(VisualStudioReporter))]
public class PngUtilTests
{
    [Fact]
    public void TestReadPng()
    {
        var image = Png.Open(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\CharTests\TestFiles\ImportChar.png"));
        TestImage(image);
    }

    [Fact]
    public void TestRead4TilesPngEvenWidth()
    {
        TestImage("4Tiles.png");
    }

    [Fact]
    public void TestRead4TilesOddWidth()
    {
        TestImage("4TilesNoEndSeparator.png");
    }

    private static void TestImage(string imageFileName)
    {
        var image = Png.Open(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, $@"..\..\..\PngTests\TestFiles\{imageFileName}"));
        TestImage(image);
    }

    private static void TestImage(Png image)
    {
        var sb = new StringBuilder();

        for (int y = 0; y < image.Height; y++)
        {
            for (int x = 0; x < image.Width; x++)
            {
                Pixel pixel = image.GetPixel(x, y);

                sb.Append($"{pixel.R:X2}{pixel.G:X2}{pixel.B:X2} ");
            }

            sb.AppendLine();
        }

        Approvals.Verify(sb.ToString());
    }
}
