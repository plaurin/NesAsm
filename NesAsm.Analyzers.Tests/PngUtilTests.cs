using ApprovalTests;
using ApprovalTests.Reporters;
using BigGustave;
using System.Text;

namespace NesAsm.Analyzers.Tests;

[UseReporter(typeof(VisualStudioReporter))]
public class PngUtilTests
{
    [Fact]
    public void TestReadPng()
    {
        var image = Png.Open(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\TestFiles\ImportChar.png"));
        TestImage(image);
    }

    [Fact]
    public void TestRead4TilesPngEvenWidth()
    {
        var image = Png.Open(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\TestFiles\Png\4Tiles.png"));
        TestImage(image);
    }

    [Fact]
    public void TestRead4TilesOddWidth()
    {
        var image = Png.Open(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, @"..\..\..\TestFiles\Png\4TilesNoEndSeparator.png"));
        TestImage(image);
    }

    private static void TestImage(Png image)
    {
        var sb = new StringBuilder();

        for (int y = 0; y < image.Height; y++)
        {
            for (int x = 0; x < image.Width; x++)
            {
                Pixel pixel = image.GetPixel(x, y);

                sb.Append($"{pixel.R:X2}{pixel.G:X2}{pixel.B:X2} ");
            }

            sb.AppendLine();
        }

        Approvals.Verify(sb.ToString());
    }
}
