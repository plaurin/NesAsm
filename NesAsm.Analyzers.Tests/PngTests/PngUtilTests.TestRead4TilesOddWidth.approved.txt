000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF 000000 000000 000000 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF FFFFFF 000000 000000 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF FFFFFF FFFFFF 000000 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF FFFFFF 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF FFFFFF FFFFFF 000000 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF FFFFFF 000000 000000 
000000 000000 000000 000000 000000 000000 000000 000000 9A0000 000000 000000 000000 000000 FFFFFF 000000 000000 000000 
9A0000 9A0000 9A0000 930000 9A0000 9A0000 9A0000 930000 B30000 9A0000 9A0000 9A0000 9A0000 9A0000 9A0000 9A0000 9A0000 
000000 000000 0D57FF 0D57FF 0D57FF 0D57FF 000000 000000 9A0000 000000 000000 000000 000000 000000 000000 000000 000000 
000000 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 000000 9A0000 000000 D21269 D21269 000000 000000 D21269 D21269 000000 
0D57FF 0D57FF 000000 0D57FF 0D57FF 000000 0D57FF 0D57FF 9A0000 000000 D21269 D21269 D21269 D21269 D21269 D21269 000000 
0D57FF 0D57FF 000000 0D57FF 0D57FF 000000 0D57FF 0D57FF 9A0000 D21269 D21269 D21269 D21269 D21269 D21269 D21269 D21269 
0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 9A0000 D21269 D21269 D21269 D21269 D21269 D21269 D21269 D21269 
0D57FF 0D57FF 000000 000000 000000 000000 0D57FF 0D57FF 9A0000 000000 D21269 D21269 D21269 D21269 D21269 D21269 000000 
000000 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 0D57FF 000000 9A0000 000000 000000 D21269 D21269 D21269 D21269 000000 000000 
000000 000000 0D57FF 0D57FF 0D57FF 0D57FF 000000 000000 9A0000 000000 000000 000000 D21269 D21269 000000 000000 000000 
